packages/mobile/src/onboarding/contacts/ImportContactsScreen.tsx

```diff
@@ -1,227 +1,187 @@
-    if (
-      prevProps.importContactsProgress.status !== ImportContactsStatus.Done &&
-      this.props.importContactsProgress.status === ImportContactsStatus.Done
-    ) {
-      // Set timeout to leave checkmark in place for a little while
-      // @ts-ignore setTimeout is incorrectly picking up the Node typings instead of RN's typings
-      this.onFinishTimeout = setTimeout(this.onFinish, 1500)
-    }
-  }
-
-  componentWillUnmount() {
-    if (this.onFinishTimeout) {
-      clearTimeout(this.onFinishTimeout)
-    }
-  }
-
-  onPressConnect = async () => {
+  async function onPressConnect() {
     CeloAnalytics.track(CustomEventNames.import_contacts)
     const hasGivenContactPermission = await requestContactsPermission()
     if (hasGivenContactPermission) {
-      this.props.importContacts(this.state.isFindMeSwitchChecked)
+      dispatch(importContacts(isFindMeSwitchChecked))
     }
   }
 
-  onPressSkip = () => {
-    this.props.cancelImportContacts()
-    this.props.denyImportContacts()
-    CeloAnalytics.track(CustomEventNames.import_contacts_skip)
-    this.onFinish()
-  }
-
-  onToggleFindMeSwitch = (value: boolean) => {
-    this.setState({
-      isFindMeSwitchChecked: value,
-    })
-  }
-
-  onFinish = () => {
+  function onFinish() {
     navigate(Screens.OnboardingSuccessScreen)
   }
 
-  renderImportStatus = () => {
-    const {
-      t,
-      importContactsProgress: { status, total },
-      matchedContacts,
-    } = this.props
-    const matchesFound = Object.keys(matchedContacts).length
+  function onToggleFindMeSwitch(value: boolean) {
+    setFindMeSwitch(value)
+  }
 
-    if (status === ImportContactsStatus.Done) {
-      return (
-        <View style={styles.statusContainer}>
-          <Checkmark />
-          {matchesFound > 0 ? (
-            <>
-              <Text style={styles.h1Status}>
-                {t('contacts.syncing.successHeader', { matches: matchesFound })}
-              </Text>
-              <Text style={styles.h2Status}>{t('contacts.syncing.successStatus')}</Text>
-            </>
-          ) : (
-            <Text style={styles.h1Status}>{t('contacts.syncing.successStatus')}</Text>
-          )}
-        </View>
-      )
+  // If import has already been done, move user forward
+  // Eventually we will want to somehow allow users to prompt matchmaking
+  React.useEffect(() => {
+    if (importStatus === ImportContactsStatus.Done) {
+      const onFinishTimeout = setTimeout(onFinish, 2000)
+      return () => {
+        clearTimeout(onFinishTimeout)
+      }
     }
+  }, [])
 
-    let h2Text: string
-    if (status === ImportContactsStatus.Processing) {
-      h2Text = 'contacts.syncing.loadingStatus1'
-    } else if (status === ImportContactsStatus.Matchmaking) {
-      h2Text = 'contacts.syncing.loadingStatus2'
-    } else {
-      h2Text = 'contacts.syncing.loadingStatus0'
+  React.useEffect(() => {
+    const prevImportContactProgress = prevImportContactProgressRef.current
```

Comment:
Does this work when the status has multiple steps? Or does the ref *always* have the first value of `importContactsProgress`?

Reply:
I believe it does. We update the ref to the latest instance on every update (line 69)

Reply:
Ah now I see. so the initial setting of prevImportContactProgressRef on line 34 is not actually the previous. Is the current.

Reply:
Yep that's right, similar to how prev props don't really exist on first render